docs/core-migration-guide.md

```diff
@@ -48,7 +48,9 @@ license: |
 
 - Since Spark 4.0, the MDC (Mapped Diagnostic Context) key for Spark task names in Spark logs has been changed from `mdc.taskName` to `task_name`. To use the key `mdc.taskName`, you can set `spark.log.legacyTaskNameMdc.enabled` to `true`.
 
-- Since Spark 4.0, Spark performs speculative executions less agressively with `spark.speculation.multiplier=3` and `spark.speculation.quantile=0.9`. To restore the legacy behavior, you can set `spark.speculation.multiplier=1.5` and `spark.speculation.quantile=0.75`.
+- Since Spark 4.0, Spark performs speculative executions less aggressively with `spark.speculation.multiplier=3` and `spark.speculation.quantile=0.9`. To restore the legacy behavior, you can set `spark.speculation.multiplier=1.5` and `spark.speculation.quantile=0.75`.
+
+- Since Spark 4.0, when configurations are provided to `spark-submit` or `spark-shell` via `--properties-file` parameter, Spark will also load extra configurations from `SPARK_CONF_DIR/spark-defaults.conf`. Those configurations defined in the former take higher precedence over those from the latter.
```

Comment:
this also changes history server/ master & worker / spark sql shell / spark thrift server/ spark connect server

Reply:
Good point. Will update.

Reply:
Actually the PR only changes `SparkSubmitArguments` so others in the above are not affected.

Reply:
spark sql shell / spark thrift server/ spark connect server is using spark-submit.

spark daemons also support starting with spark-class or spark-submit mode.

Reply:
How about adding one more sentence: "Other components such as Spark thrift server, connect server, SQL shell that call `spark-submit` internally will also have the same behavior change"?

Reply:
spark sql shell? If you mean `bin/spark-sql`, it uses `bin/spark-submit` script but using the script doesn't mean it uses `SparkSubmit` and `SparkSubmitArguments`.

Reply:
I'm also curious where do Spark thrift server and others use `SparkSubmit` and `SparkSubmitArguments`..

Reply:
Yeah. I mean `bin/spark-sql` but it goes through `SparkSubmit` and `SparkSubmitArguments` if it uses `spark-submit` with a launcher proc

Reply:
https://github.com/apache/spark/blob/79aeae1a9aaa2e9cfaf03a1f5d88e1447a3f9b19/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L76

Reply:
Oh right,`SparkSubmit` will load specified class which is given as parameter to `spark-submit`.

Reply:
> How about adding one more sentence: "Other components such as Spark thrift server, connect server, SQL shell that call `spark-submit` internally will also have the same behavior change"?

Or, just mention any commands which take `--properties-file` parameter, e.g., SQL shell, connect server, etc., all have this change?

Reply:
This may not be true for all cases. For instance, Spark history server also takes `--properties-file` but does not go through `SparkSubmit` and `SparkSubmitArguments`.